### ISSUE TITLE
Using docker executor, can non-primary containers communicate with each other?

### ISSUE THREAD

**Issue by thomasmatecki** [circleci/circleci-docs#7729]
# Missing Information
If using multiple containers with the docker executor, can non-primary containers communicate with each other?

Docs currently state:
> It is possible to specify multiple images for your job. Specify multiple images if, for example, you need to use a database for your tests or for some other required service. All containers run in a common network and every exposed port will be available on localhost from a [primary container](https://circleci.com/docs/glossary/#primary-container).

If I am running cypress tests on an application that has a static frontend, an  API server, a database and also some other stuff (let's say Redis).

```
Cypress (chrome) | -> API container (e.g. django) | -> Database Container
                 |                                | -> Redis Container
                 | -> Static UI (e.g React)

```

... I need a least two containers that can access another container, so network connectivity from only the primary container doesn't really work.

If using docker compose (for example), this is accomplished by using services names as if they are hostnames. The docs don't say that I can do that though. Would that work?

# Already Looked
[Using multiple Docker images](https://circleci.com/docs/using-docker/#using-multiple-docker-images)

# Potential Locations
Same as above: [Using multiple Docker images](https://circleci.com/docs/using-docker/#using-multiple-docker-images)

**Comment by rosieyohannan** [comment]
Thanks for your question! I have checked with engineering and they explained that yes, all containers can talk to each other. When using multiple Docker containers for a job, the containers share a cgroups network namespace. So, in terms of networking, it's like every container shares the same localhost. They also share the same process namespace, so each can see the other's processes, too. Hopefully this is helpful, please let me know either way and I can try and expand on this.

**Comment by rosieyohannan** [comment]
Yes, running two services on the same port would cause the pipeline to fail.

Let me know how you get on, and yes I'll make an update to the page to clarify.
